utils/calc_total_unclaimed_fees.go

```
@@ -0,0 +1,54 @@
+package utils
+
+import (
+	"log"
+	"math/big"
+	"strconv"
+)
+
+var tokenSymbolToName = map[string]string{
+	"WETH": "ethereum",
+	"DAI":  "dai",
+	"UNI":  "uniswap",
+}
+
+// calculates total unclaimed fees for the pool
+func CalcTotalUnclaimedFees(
+	token0_unclaimed_fees string,
+	token1_unclaimed_fees string,
+	token0_name string,
+	token1_name string,
+) string {
+	// Convert unclaimed fees for each token to *big.Float
+	token0_float := new(big.Float)
+	token1_float := new(big.Float)
+	token0_float.SetString(token0_unclaimed_fees)
+	token1_float.SetString(token1_unclaimed_fees)
+
+	// Fetch current prices of each token
+	token0PriceInDollars, err := FetchTokenPrice(tokenSymbolToName[token0_name])
+	token1PriceInDollars, err := FetchTokenPrice(tokenSymbolToName[token1_name])
+	if err != nil {
+		log.Println(err)
```

All errors in this func should return the error